tensorflow/java/src/main/java/org/tensorflow/Input.java

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow;
+
+/**
+ * A tensor that can be used as an operand for operation wrappers.
+ *
+ * <p>The {@code Input} interface is at the base of the Java Ops API by allowing any object that
+ * wraps an input tensor to be passed as an operand to an operation.
+ *
+ * <pre>{@code
+ * // Output implements Input.
+ * Output image = ops.image().decodeJpeg(...).image();
+ * ops.math().cast(image, DataType.FLOAT);
+ *
+ * // DecodeJpeg operation also implements Input, returning the image as its output.
+ * ops.math().cast(ops.image().decodeJpeg(...), DataType.FLOAT);
+ * }</pre>
+ *
+ * <p>Additionally, objects implementing {@code Iterable<Input>} can be passed in input to
+ * operations handling a list of tensors as one of its operand.
+ *
+ * <pre>{@code
+ * // List<Output> extends from Iterable<Input>
+ * List<Output> outputs = ops.array().split(...).output();
```

Comment:
I see you prefer to use the named outputs, but we also plan to support the operations directly.
If so, I feel it's better to use just the op name (instead of naming the output) when there is a single output. So `split(...)` instead of `split(...).output()`. What do you think?